src/actions/schemes.js

```diff
@@ -40,17 +44,30 @@ export function createScheme(scheme: Scheme) {
     );
 }
 
-export function deleteScheme(schemeId: string) {
-    return bindClientFunc(
-        Client4.deleteScheme,
-        SchemeTypes.DELETE_SCHEME_REQUEST,
-        [SchemeTypes.DELETED_SCHEME, SchemeTypes.DELETE_SCHEME_SUCCESS],
-        SchemeTypes.DELETE_SCHEME_FAILURE,
-        schemeId
-    );
+export function deleteScheme(schemeId: string): ActionFunc {
+    return async (dispatch, getState) => {
+        dispatch({type: SchemeTypes.DELETE_SCHEME_REQUEST, data: null}, getState);
+
+        let data = null;
+        try {
+            data = await Client4.deleteScheme(schemeId);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(batchActions([
+                {type: SchemeTypes.DELETE_SCHEME_FAILURE, error},
+                logError(error)(dispatch),
+            ]), getState);
+            return {data: null, error};
+        }
+
+        dispatch({type: SchemeTypes.DELETED_SCHEME, data: {schemeId}}, getState);
```

Comment:
Here is the real change. Here we send the schemeId in the DELETE_SCHEME event to properly remove the scheme from the store on success.